lib/web/command_test.go

```diff
@@ -218,14 +218,38 @@ func waitForCommandOutput(stream io.Reader, substr string) error {
 	}
 }
 
-// Test_runCommands tests that runCommands runs the given command on all hosts.
-// The commands should run in parallel, but we don't have a deterministic way to
-// test that (sleep with checking the execution time in not deterministic).
+// Test_runCommands tests that runCommands runs the given command on all hosts,
+// and that a given number of executions are made in parallel.
 func Test_runCommands(t *testing.T) {
```

Comment:
This implementation has a chance of being flaky, so it's up to y'all whether it's worth testing the parallelism 🤷

Reply:
Why do we need to test this in this way? I think there are two things that we want to test here
1. The `workers number` is set correctly
2. The run command runs as many times as expected.

IMO that fact runs in parallel in an implementation detail, and we recently removed a bunch of tests that tried to make some assertions about parallel running by using timeouts because all of them were flaky. I think we should revert this test to the previous state and add the test that covers the first scenario (we could also add a test that sets the 'num_workers` to the default value if the value is not provided).